File: HotelReservationServiceJava/RoomTypeD.java
//This room only has a minimum duration of days
public class RoomTypeD extends Room
{

    int minimumDuration;
    
    public RoomTypeD(int capacity, double perPersonPrice, int minimumDuration){
        super(capacity, perPersonPrice);
        this.minimumDuration = minimumDuration;
    }
    
    //check duration and propagate logic to Room.class
    @Override
    public boolean addReservation(Reservation res){
        if(res.getDuration() >= minimumDuration){
            return super.addReservation(res);
        }
        return false;
    }
    
}

File: HotelReservationServiceCpp/src/RoomTypeD.cpp
#include "RoomTypeD.h"

RoomTypeD::RoomTypeD(int cap, double ppPrice, int mDuration) :
    minimumDuration{mDuration},
    Room{cap, ppPrice}
{}

bool RoomTypeD::addReservation(Reservation& reservation){
    if(reservation.getDuration() >= minimumDuration){
        return Room::addReservation(reservation);
    }
    return false;
}

RoomTypeD::~RoomTypeD(){}
